pkg/cmd/cli/config/smart_merge.go

```diff
@@ -67,42 +124,28 @@ func MergeConfig(haystack, addition clientcmdapi.Config) (*clientcmdapi.Config,
 	ret := haystack
 
 	requestedClusterNamesToActualClusterNames := map[string]string{}
-	existingClusterNames, err := getMapKeys(reflect.ValueOf(haystack.Clusters))
-	if err != nil {
-		return nil, err
-	}
 	for requestedKey, needle := range addition.Clusters {
 		if existingName := FindExistingClusterName(haystack, needle); len(existingName) > 0 {
 			requestedClusterNamesToActualClusterNames[requestedKey] = existingName
 			continue
 		}
 
-		uniqueName := getUniqueName(requestedKey, existingClusterNames)
-		requestedClusterNamesToActualClusterNames[requestedKey] = uniqueName
-		ret.Clusters[uniqueName] = needle
+		requestedClusterNamesToActualClusterNames[requestedKey] = requestedKey
+		ret.Clusters[requestedKey] = needle
 	}
 
 	requestedAuthInfoNamesToActualAuthInfoNames := map[string]string{}
-	existingAuthInfoNames, err := getMapKeys(reflect.ValueOf(haystack.AuthInfos))
-	if err != nil {
-		return nil, err
-	}
 	for requestedKey, needle := range addition.AuthInfos {
 		if existingName := FindExistingAuthInfoName(haystack, needle); len(existingName) > 0 {
```

Comment:
do we still want to scan authinfo stanzas, or only use the stanza name getUserNickname would produce?